presto-spark-base/src/main/java/com/facebook/presto/spark/execution/AbstractPrestoSparkQueryExecution.java

```diff
@@ -708,6 +717,144 @@ else if (writerTarget instanceof ExecutionWriterTarget.RefreshMaterializedViewHa
         }
     }
 
+    // Returns RDD for specified fragmented subplan
+    // This method ensures that RDD is created only once for a sub-plan, where identity is determined by fragment id
+    protected <T extends PrestoSparkTaskOutput> RddAndMore<T> createRddFragmented(SubPlan subPlan, TableWriteInfo tableWriteInfo)
+            throws SparkException, TimeoutException
+    {
+        synchronized (fragmentIdToRdd) {
+            if (fragmentIdToRdd.containsKey(subPlan.getFragment().getId())) {
+                return fragmentIdToRdd.get(subPlan.getFragment().getId());
+            }
+
+            ImmutableMap.Builder<PlanFragmentId, JavaPairRDD<MutablePartitionId, PrestoSparkMutableRow>> rddInputs = ImmutableMap.builder();
```

Comment:
except the part were we get existing rdds from the map and add this back to the map, most of the logic isn't special for AQE.  Instead of dupplicating, it should be shared with static executor

Reply:
Lets discuss this. My understanding was that we wanted to get the building blocks out first and then go for refactoring in parallel.

Reply:
I think extracting some common methods for these cases should be done as part of this PR.